File: WebApi/angular/src/app/shared/model/LoanCalculationDataVM.ts
import { DictionaryRecordVM } from '@loancalc/shared/model';

export interface ILoanCalculationDataVM {
    TotalValueOfLoan: number;   
    NumberOfAnnualInstallments: number; 
    AnnualInterestInstallments: number;
    Currency: DictionaryRecordVM; 
}
export class LoanCalculationDataVM implements ILoanCalculationDataVM {
    TotalValueOfLoan: number;   
    NumberOfAnnualInstallments: number; 
    AnnualInterestInstallments: number;
    Currency: DictionaryRecordVM; 
}

File: WebApi/Models/Loan/LoanCalculationDataVM.cs
namespace WebApi.Models.Loan
{
    public class LoanCalculationDataVM
    {
        public decimal TotalValueOfLoan { get; set; }
        public int NumberOfAnnualInstallments { get; set; }
        public decimal AnnualInterestInstallments { get; set; }
        public DictionaryRecordVM Currency { get; set; }
    }
}